// vue3-material/dist/vue3-material.ssr.js
'use strict';var vue=require('vue');function _slicedToArray(arr, i) {
  return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest();
}

function _toConsumableArray(arr) {
  return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread();
}

function _arrayWithoutHoles(arr) {
  if (Array.isArray(arr)) return _arrayLikeToArray(arr);
}

function _arrayWithHoles(arr) {
  if (Array.isArray(arr)) return arr;
}

function _iterableToArray(iter) {
  if (typeof Symbol !== "undefined" && Symbol.iterator in Object(iter)) return Array.from(iter);
}

function _iterableToArrayLimit(arr, i) {
  if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return;
  var _arr = [];
  var _n = true;
  var _d = false;
  var _e = undefined;

  try {
    for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) {
      _arr.push(_s.value);

      if (i && _arr.length === i) break;
    }
  } catch (err) {
    _d = true;
    _e = err;
  } finally {
    try {
      if (!_n && _i["return"] != null) _i["return"]();
    } finally {
      if (_d) throw _e;
    }
  }

  return _arr;
}

function _unsupportedIterableToArray(o, minLen) {
  if (!o) return;
  if (typeof o === "string") return _arrayLikeToArray(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor) n = o.constructor.name;
  if (n === "Map" || n === "Set") return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen);
}

function _arrayLikeToArray(arr, len) {
  if (len == null || len > arr.length) len = arr.length;

  for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i];

  return arr2;
}

function _nonIterableSpread() {
  throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}

function _nonIterableRest() {
  throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}// vue
var script = vue.defineComponent({
  props: {
    type: {
      type: String,
      default: 'button'
    },
    background: {
      type: String,
      default: '#3ca354'
    },
    color: {
      type: String,
      default: '#ffffff'
    },
    elevation: {
      type: Number,
      default: 1
    },
    icon: {
      type: String,
      default: 'fal fa-coffee'
    },
    isSmall: {
      type: Boolean,
      default: false
    }
  },
  // eslint-disable-next-line
  setup: function setup(props) {
    // refs
    var btnRipple = vue.ref(null); // methods

    var createRipple = function createRipple(event) {
      var button = event.currentTarget;

      if (button) {
        var circle = document.createElement('span');
        var diameter = Math.max(button.clientWidth, button.clientHeight);
        var radius = diameter / 2;
        circle.style.width = circle.style.height = "".concat(diameter, "px");
        var buttonElement = button.getBoundingClientRect();
        circle.style.left = "\n                    ".concat(event.clientX - buttonElement.left - radius, "px\n                ");
        circle.style.top = "\n                    ".concat(event.clientY - buttonElement.top - radius, "px\n                ");
        circle.classList.add('ripple');
        var ripple = button.getElementsByClassName('ripple')[0];

        if (ripple) {
          ripple.remove();
        }

        button.appendChild(circle);
      }
    }; // lifecycle hooks


    vue.onMounted(function () {
      if (btnRipple.value) {
        btnRipple.value.style.setProperty('--background', props.background);
        var elevationShadowColor = props.elevation * 0.5;
        btnRipple.value.style.setProperty('--elevation-shadow', "0 2px 4px rgba(0, 0, 0, ".concat(elevationShadowColor, ")"));
        btnRipple.value.style.setProperty('--color', props.color);
      }
    });
    return {
      props: props,
      createRipple: createRipple,
      btnRipple: btnRipple
    };
  }
});var _withId = /*#__PURE__*/vue.withScopeId("data-v-1bb76b2c");

vue.pushScopeId("data-v-1bb76b2c");

var _hoisted_1 = /*#__PURE__*/vue.createVNode("div", {
  class: "btn-overlay"
}, null, -1);

vue.popScopeId();

var render = /*#__PURE__*/_withId(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("button", {
    class: ["btn-ripple", {
      small: _ctx.isSmall
    }],
    onMousedown: _cache[1] || (_cache[1] = function () {
      return _ctx.createRipple && _ctx.createRipple.apply(_ctx, arguments);
    }),
    type: _ctx.type,
    ref: "btnRipple"
  }, [_hoisted_1, vue.createVNode("i", {
    class: _ctx.props.icon
  }, null, 2)], 42, ["type"]);
});function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}var css_248z = "\n.btn-ripple[data-v-1bb76b2c] {\n    --size: 45px;\n    width: var(--size);\n    height: var(--size);\n    min-width: var(--size);\n    min-height: var(--size);\n    max-width: var(--size);\n    max-height: var(--size);\n    user-select: none;\n    display: inline-block;\n    position: relative;\n    overflow: hidden;\n    transition: background 400ms;\n    color: var(--color);\n    background: var(--background);\n    font-size: 1.6rem;\n    outline: 0;\n    border: 0;\n    border-radius: calc(var(--size) / 2);\n    box-shadow: var(--elevation-shadow);\n    cursor: pointer;\n}\n.btn-ripple:hover .btn-overlay[data-v-1bb76b2c] {\n    background: rgba(255, 255, 255, 0.05);\n}\n.btn-ripple:focus .btn-overlay[data-v-1bb76b2c] {\n    background: rgba(255, 255, 255, 0.1);\n}\n.small[data-v-1bb76b2c] {\n    --size: 35px;\n    font-size: 1.4rem;\n}\n.btn-overlay[data-v-1bb76b2c] {\n    position: absolute;\n    top: 0;\n    left: 0;\n    right: 0;\n    bottom: 0;\n    background: rgba(255, 255, 255, 0);\n    transition-duration: 0.1s;\n}\n";
styleInject(css_248z);var css_248z$1 = "\nspan.ripple {\n    display: flex;\n    position: absolute;\n    border-radius: 50%;\n    transform: scale(0);\n    animation: ripple 600ms linear;\n    background: rgba(255, 255, 255, 0.7);\n    box-shadow: 0 0 10px 10px rgba(255, 255, 255, 0.7);\n}\n@keyframes ripple {\nto {\n        transform: scale(4);\n        opacity: 0;\n}\n}\n";
styleInject(css_248z$1);script.render = render;
script.__scopeId = "data-v-1bb76b2c";// vue
var script$1 = vue.defineComponent({
  props: {
    elevation: {
      type: Number,
      default: 2
    }
  },
  // eslint-disable-next-line
  setup: function setup(props) {
    return {
      props: props
    };
  }
});var _withId$1 = /*#__PURE__*/vue.withScopeId("data-v-2ddee2f9");

var render$1 = /*#__PURE__*/_withId$1(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", {
    class: ["app-bar", ["elevation-".concat(_ctx.props.elevation), "elevation-background-".concat(_ctx.props.elevation)]]
  }, [vue.renderSlot(_ctx.$slots, "default")], 2);
});var css_248z$2 = "\n.app-bar[data-v-2ddee2f9] {\n    min-height: 4rem;\n    display: flex;\n    width: 100%;\n}\n";
styleInject(css_248z$2);script$1.render = render$1;
script$1.__scopeId = "data-v-2ddee2f9";// vue
var script$2 = vue.defineComponent({
  props: {
    variant: {
      type: String,
      validator: function validator(prop) {
        return ['contained', 'outlined', 'text'].includes(prop);
      },
      default: 'contained'
    },
    routerPath: {
      type: String,
      default: ''
    },
    isDisabled: {
      type: Boolean,
      default: false
    },
    isSubmit: {
      type: Boolean,
      default: false
    },
    primaryColor: {
      type: String,
      default: '98, 0, 238'
    },
    elevation: {
      type: Number,
      default: 2
    },
    isAutoTextColor: {
      type: Boolean,
      default: false
    }
  },
  // eslint-disable-next-line
  setup: function setup(props) {
    // refs
    var btnRipple = vue.ref(null); // methods

    var lightOrDark = function lightOrDark(color) {
      var rgbArray = color.split(',');

      if (rgbArray.length !== 3) {
        return 'light';
      }

      var r = parseInt(rgbArray[0]);
      var g = parseInt(rgbArray[1]);
      var b = parseInt(rgbArray[2]); // HSP equation from http://alienryderflex.com/hsp.html

      var hsp = Math.sqrt(0.299 * (r * r) + 0.587 * (g * g) + 0.114 * (b * b));

      if (hsp > 127.5) {
        return 'light';
      } else {
        return 'dark';
      }
    }; // computed


    var cssProperties = vue.computed(function () {
      return "--primary-color: ".concat(props.primaryColor, ";") + "--text-color: ".concat(props.isAutoTextColor ? lightOrDark(props.primaryColor) === 'light' ? '0, 0, 0' : '255, 255, 255' : '255, 255, 255', ";") + "--ripple-color: ".concat(props.variant !== 'contained' ? props.primaryColor : props.isAutoTextColor ? lightOrDark(props.primaryColor) === 'light' ? '0, 0, 0' : '255, 255, 255' : '255, 255, 255', ";") + "--default-elevation: var(--elevation-".concat(props.elevation, ");") + "--focus-elevation: var(--elevation-".concat(props.elevation + 2, ");") + "--active-elevation: var(--elevation-".concat(props.elevation + 6, ");");
    }); // methods

    var createRipple = function createRipple(event) {
      var button = event.currentTarget;

      if (button) {
        var circle = document.createElement('span');
        var diameter = Math.max(button.clientWidth, button.clientHeight);
        var radius = diameter / 2;
        circle.style.width = circle.style.height = "".concat(diameter, "px");
        var buttonElement = button.getBoundingClientRect();
        circle.style.left = "\n                    ".concat(event.clientX - buttonElement.left - radius, "px\n                ");
        circle.style.top = "\n                    ".concat(event.clientY - buttonElement.top - radius, "px\n                ");
        circle.classList.add('ripple');

        if (btnRipple.value) {
          var ripple = button.getElementsByClassName('ripple')[0];

          if (ripple) {
            ripple.remove();
          }
        }

        button.appendChild(circle);
      }
    };

    return {
      props: props,
      createRipple: createRipple,
      btnRipple: btnRipple,
      cssProperties: cssProperties
    };
  }
});var _withId$2 = /*#__PURE__*/vue.withScopeId("data-v-3dcf3104");

vue.pushScopeId("data-v-3dcf3104");

var _hoisted_1$1 = /*#__PURE__*/vue.createVNode("div", {
  class: "btn-overlay"
}, null, -1);

var _hoisted_2 = {
  class: "content"
};

vue.popScopeId();

var render$2 = /*#__PURE__*/_withId$2(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock(vue.resolveDynamicComponent(_ctx.routerPath ? 'router-link' : 'button'), {
    class: ["btn-ripple", ["btn-".concat(_ctx.props.variant), {
      'disabled': _ctx.isDisabled
    }]],
    style: _ctx.cssProperties,
    onMousedown: _ctx.createRipple,
    type: _ctx.isSubmit ? 'submit' : 'button',
    ref: "btnRipple",
    to: _ctx.routerPath
  }, {
    default: _withId$2(function () {
      return [_hoisted_1$1, vue.createVNode("div", _hoisted_2, [vue.renderSlot(_ctx.$slots, "default")])];
    }),
    _: 3
  }, 8, ["class", "style", "onMousedown", "type", "to"]);
});var css_248z$3 = "\n/* contained */\n.btn-contained[data-v-3dcf3104] {\n    color: rgb(var(--text-color));\n    background: rgb(var(--primary-color));\n    box-shadow: var(--default-elevation);\n    border: 0;\n    padding: 10px 16px;\n}\n.btn-overlay[data-v-3dcf3104] {\n    position: absolute;\n    top: 0;\n    left: 0;\n    right: 0;\n    bottom: 0;\n    background: rgba(255, 255, 255, 0);\n    transition-duration: 0.1s;\n}\n.btn-contained:hover .btn-overlay[data-v-3dcf3104] {\n    background: rgba(255, 255, 255, 0.05);\n}\n.btn-contained:focus .btn-overlay[data-v-3dcf3104] {\n    background: rgba(255, 255, 255, 0.15);\n}\n\n/* outlined and text */\n.btn-outlined[data-v-3dcf3104] {\n    background: none;\n    color: rgb(var(--primary-color));\n    border: 1px solid rgb(var(--primary-color));\n    padding: 10px 15px;\n}\n.btn-text[data-v-3dcf3104] {\n    background: none;\n    color: rgb(var(--primary-color));\n    border: 0;\n    padding: 10px 16px;\n}\n.btn-text[data-v-3dcf3104]:hover,\n.btn-outlined[data-v-3dcf3104]:hover {\n    background: rgba(var(--primary-color), 0.1);\n}\n.btn-text[data-v-3dcf3104]:focus,\n.btn-outlined[data-v-3dcf3104]:focus {\n    background: rgba(var(--primary-color), 0.2);\n}\n\n/* content */\n.content[data-v-3dcf3104] {\n    z-index: 10;\n}\n\n/* ripple */\n.btn-ripple[data-v-3dcf3104] {\n    font-family: 'Roboto', sans-serif;\n    user-select: none;\n    display: inline-flex;\n    align-items: center;\n    position: relative;\n    overflow: hidden;\n    font-size: 0.88rem;\n    font-weight: 500;\n    letter-spacing: 1.25px;\n    outline: 0;\n    border-radius: 4px;\n    cursor: pointer;\n    white-space: nowrap;\n    box-sizing: border-box;\n    text-decoration: none;\n    transition: background 100ms, box-shadow 300ms;\n}\n.btn-contained[data-v-3dcf3104]:hover,\n.btn-contained[data-v-3dcf3104]:focus {\n    box-shadow: var(--focus-elevation);\n}\n.btn-contained[data-v-3dcf3104]:active:hover {\n    box-shadow: var(--active-elevation);\n}\n.disabled[data-v-3dcf3104] {\n    opacity: 0.5;\n    pointer-events: none;\n}\n";
styleInject(css_248z$3);var css_248z$4 = "\nspan.ripple {\n    --visible-opacity: 0.2;\n    pointer-events: none;\n    display: flex;\n    position: absolute;\n    border-radius: 50%;\n    pointer-events: none;\n    background-color: rgba(var(--ripple-color), 0);\n    box-shadow: 0 0 5px 5px rgba(var(--ripple-color), var(--visible-opacity));\n    animation: ripple 450ms ease-in;\n    transform: scale(4);\n    transition: background-color 0.2s;\n}\n.btn-ripple:active:hover span.ripple {\n    background-color: rgba(\n        var(--ripple-color),\n        var(--visible-opacity)\n    ) !important;\n}\n@keyframes ripple {\n0% {\n        background-color: rgba(var(--ripple-color), var(--visible-opacity));\n        transform: scale(0);\n}\n66% {\n        background-color: rgba(var(--ripple-color), var(--visible-opacity));\n        transform: scale(4);\n}\n100% {\n        background-color: rgba(var(--ripple-color), 0);\n}\n}\n";
styleInject(css_248z$4);script$2.render = render$2;
script$2.__scopeId = "data-v-3dcf3104";// vue
var script$3 = vue.defineComponent({
  props: {
    elevation: {
      type: Number,
      default: 2
    },
    borderRadius: {
      type: Number,
      default: 5
    }
  },
  // eslint-disable-next-line
  setup: function setup(props) {
    // computed
    var cssProperties = vue.computed(function () {
      return "--card-border-radius: ".concat(props.borderRadius, "px;");
    });
    return {
      props: props,
      cssProperties: cssProperties
    };
  }
});var _withId$3 = /*#__PURE__*/vue.withScopeId("data-v-114efd9e");

var render$3 = /*#__PURE__*/_withId$3(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", {
    class: ["card", ["elevation-".concat(_ctx.props.elevation), "elevation-background-".concat(_ctx.props.elevation)]],
    style: _ctx.cssProperties
  }, [vue.renderSlot(_ctx.$slots, "default")], 6);
});var css_248z$5 = "\n.card[data-v-114efd9e] {\n    position: relative;\n    border-radius: var(--card-border-radius);\n    overflow: hidden;\n    transition: box-shadow 0.2s;\n}\n";
styleInject(css_248z$5);script$3.render = render$3;
script$3.__scopeId = "data-v-114efd9e";// vue
var script$4 = vue.defineComponent({
  // eslint-disable-next-line
  setup: function setup(props) {
    return {
      props: props
    };
  }
});var _withId$4 = /*#__PURE__*/vue.withScopeId("data-v-e2fc2772");

vue.pushScopeId("data-v-e2fc2772");

var _hoisted_1$2 = {
  class: "card-content"
};

vue.popScopeId();

var render$4 = /*#__PURE__*/_withId$4(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", _hoisted_1$2, [vue.renderSlot(_ctx.$slots, "default")]);
});var css_248z$6 = "\n.card-content[data-v-e2fc2772] {\n    padding: 15px 20px;\n}\n";
styleInject(css_248z$6);script$4.render = render$4;
script$4.__scopeId = "data-v-e2fc2772";// vue
var script$5 = vue.defineComponent({
  // eslint-disable-next-line
  setup: function setup(props) {
    return {
      props: props
    };
  }
});var _withId$5 = /*#__PURE__*/vue.withScopeId("data-v-4bb1fe1a");

vue.pushScopeId("data-v-4bb1fe1a");

var _hoisted_1$3 = {
  class: "card-header"
};

vue.popScopeId();

var render$5 = /*#__PURE__*/_withId$5(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", _hoisted_1$3, [vue.renderSlot(_ctx.$slots, "default")]);
});var css_248z$7 = "\n.card-header[data-v-4bb1fe1a] {\n    padding: 15px 20px;\n    font-size: 1.2rem;\n    font-weight: 500;\n}\n";
styleInject(css_248z$7);script$5.render = render$5;
script$5.__scopeId = "data-v-4bb1fe1a";// vue
var script$6 = vue.defineComponent({
  setup: function setup() {
    return {};
  }
});var _withId$6 = /*#__PURE__*/vue.withScopeId("data-v-bebb5804");

vue.pushScopeId("data-v-bebb5804");

var _hoisted_1$4 = {
  class: "container"
};
var _hoisted_2$1 = {
  class: "content"
};

vue.popScopeId();

var render$6 = /*#__PURE__*/_withId$6(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", _hoisted_1$4, [vue.createVNode("div", _hoisted_2$1, [vue.renderSlot(_ctx.$slots, "default")])]);
});var css_248z$8 = "\n.container[data-v-bebb5804] {\n    display: flex;\n    justify-content: center;\n    padding: 0 calc(var(--default-margin) * 4);\n}\n.content[data-v-bebb5804] {\n    flex: 1;\n    max-width: 1200px;\n    width: 100%;\n}\n";
styleInject(css_248z$8);script$6.render = render$6;
script$6.__scopeId = "data-v-bebb5804";// vue
var script$7 = vue.defineComponent({
  props: {
    spacing: {
      type: Number,
      default: 50
    }
  },
  // eslint-disable-next-line
  setup: function setup(props) {
    // computed
    var cssProperties = vue.computed(function () {
      return "--spacing: ".concat(props.spacing, "px;");
    });
    return {
      cssProperties: cssProperties
    };
  }
});var _withId$7 = /*#__PURE__*/vue.withScopeId("data-v-5c14f881");

var render$7 = /*#__PURE__*/_withId$7(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", {
    class: "divider",
    style: _ctx.cssProperties
  }, null, 4);
});var css_248z$9 = "\n.divider[data-v-5c14f881] {\n    width: 100%;\n    padding-bottom: var(--spacing);\n    margin-bottom: var(--spacing);\n    border-bottom: 1px solid rgba(var(--default-color), 0.1);\n}\n";
styleInject(css_248z$9);script$7.render = render$7;
script$7.__scopeId = "data-v-5c14f881";var script$8 = vue.defineComponent({
  props: {
    modelValue: {
      type: Number,
      default: 0
    },
    label: {
      type: String,
      default: 'Default'
    },
    placeholder: {
      type: String,
      default: 'Select an option'
    },
    options: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    value: {
      type: String,
      default: 'id'
    },
    title: {
      type: String,
      default: 'title'
    },
    sort: {
      type: Function,
      default: function _default() {
        return true;
      }
    }
  },
  // eslint-disable-next-line
  setup: function setup(props, context) {
    // refs
    var dropdownField = vue.ref(null); // computed

    var isInputFilled = vue.computed(function () {
      return props.modelValue > 0;
    });
    var sortedOptions = vue.computed(function () {
      var tmpArray = _toConsumableArray(props.options);

      return tmpArray.sort(props.sort);
    }); // methods

    var updateValue = function updateValue(e) {
      var val = parseInt(e.target.value);
      context.emit('update:modelValue', val);
    };

    return {
      props: props,
      sortedOptions: sortedOptions,
      isInputFilled: isInputFilled,
      dropdownField: dropdownField,
      updateValue: updateValue
    };
  }
});var _withId$8 = /*#__PURE__*/vue.withScopeId("data-v-15f27a7b");

vue.pushScopeId("data-v-15f27a7b");

var _hoisted_1$5 = {
  class: "dropdown"
};
var _hoisted_2$2 = {
  class: "default-option-text",
  value: 0
};

var _hoisted_3 = /*#__PURE__*/vue.createVNode("div", {
  class: "dropdown-left"
}, null, -1);

var _hoisted_4 = {
  class: "dropdown-notch"
};
var _hoisted_5 = {
  class: "dropdown-label"
};

var _hoisted_6 = /*#__PURE__*/vue.createVNode("div", {
  class: "dropdown-right"
}, null, -1);

vue.popScopeId();

var render$8 = /*#__PURE__*/_withId$8(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", _hoisted_1$5, [vue.createVNode("select", {
    class: ["dropdown-field", {
      'default-option': !_ctx.isInputFilled
    }],
    ref: "dropdownField",
    value: _ctx.modelValue,
    onInput: _cache[1] || (_cache[1] = function () {
      return _ctx.updateValue && _ctx.updateValue.apply(_ctx, arguments);
    })
  }, [vue.createVNode("option", _hoisted_2$2, vue.toDisplayString(_ctx.props.placeholder), 1), (vue.openBlock(true), vue.createBlock(vue.Fragment, null, vue.renderList(_ctx.sortedOptions, function (option, i) {
    return vue.openBlock(), vue.createBlock("option", {
      key: i,
      value: option[_ctx.props.value]
    }, vue.toDisplayString(option[_ctx.props.title]), 9, ["value"]);
  }), 128))], 42, ["value"]), _hoisted_3, vue.createVNode("div", _hoisted_4, [vue.createVNode("div", _hoisted_5, vue.toDisplayString(_ctx.props.label), 1)]), _hoisted_6]);
});var css_248z$a = "\n.dropdown[data-v-15f27a7b] {\n    --font-size: 1.1rem;\n    --label-padding: 6px;\n    --horizontal-padding: 16px;\n    --vertical-padding: 16px;\n    --border: 1px solid rgba(255, 255, 255, 0.3);\n    --height: calc(var(--font-size) + (var(--vertical-padding) * 2) + 2px);\n    display: flex;\n    width: 100%;\n    font-size: 1.1rem;\n    border-radius: 4px;\n    background: rgba(255, 255, 255, 0);\n    position: relative;\n    color: #fff;\n}\n.dropdown-left[data-v-15f27a7b],\n.dropdown-notch[data-v-15f27a7b],\n.dropdown-right[data-v-15f27a7b] {\n    min-height: var(--height);\n    max-height: var(--height);\n    padding: var(--vertical-padding) 0;\n}\n.dropdown-left[data-v-15f27a7b] {\n    padding-left: calc(var(--horizontal-padding) - var(--label-padding));\n    border: var(--border);\n    border-right: none;\n    border-top-left-radius: 4px;\n    border-bottom-left-radius: 4px;\n}\n.dropdown-notch[data-v-15f27a7b] {\n    display: flex;\n    align-items: center;\n    padding: var(--vertical-padding) 0;\n    border-bottom: var(--border);\n}\n.dropdown-right[data-v-15f27a7b] {\n    width: 100%;\n    border: var(--border);\n    border-left: none;\n    border-top-right-radius: 4px;\n    border-bottom-right-radius: 4px;\n}\n.dropdown-field[data-v-15f27a7b] {\n    flex: 1;\n    width: 100%;\n    height: 100%;\n    position: absolute;\n    font-size: 1.1rem;\n    border: none;\n    padding: 0 calc(var(--horizontal-padding) - 4px);\n    background: rgba(255, 255, 255, 0);\n    color: #fff;\n    margin-right: var(--horizontal-padding);\n}\n.dropdown-label[data-v-15f27a7b] {\n    flex: 1;\n    padding: 0 6px;\n    color: #aaa;\n    transition-duration: 0.2s;\n    transform: translateY(calc((-1 / 2) * var(--height)));\n    font-size: 0.8rem;\n    white-space: nowrap;\n}\n.dropdown-field option[data-v-15f27a7b] {\n    color: #000;\n}\n.default-option[data-v-15f27a7b] {\n    color: #aaa;\n}\n.dropdown-field option.default-option-text[data-v-15f27a7b] {\n    color: #777;\n}\n.dropdown-field:focus ~ .dropdown-left[data-v-15f27a7b],\n.dropdown-field:focus ~ .dropdown-notch[data-v-15f27a7b],\n.dropdown-field:focus ~ .dropdown-right[data-v-15f27a7b] {\n    border-color: #2c60d1;\n}\n.dropdown-field:focus ~ .dropdown-notch .dropdown-label[data-v-15f27a7b] {\n    color: #82a4ed;\n}\n";
styleInject(css_248z$a);script$8.render = render$8;
script$8.__scopeId = "data-v-15f27a7b";// vue
var script$9 = vue.defineComponent({
  props: {
    scale: {
      type: Number,
      default: 1
    },
    color: {
      type: String,
      default: 'rgb(145 201 82)'
    },
    ringWidth: {
      type: Number,
      default: 6
    },
    width: {
      type: Number,
      default: 80
    },
    height: {
      type: Number,
      default: 80
    }
  },
  // eslint-disable-next-line
  setup: function setup(props) {
    // refs
    var loadingSpinner = vue.ref(null); // lifecycle hooks

    vue.onMounted(function () {
      if (loadingSpinner.value) {
        loadingSpinner.value.style.setProperty('--spinner-scale', props.scale + '');
        loadingSpinner.value.style.setProperty('--spinner-color', props.color);
        loadingSpinner.value.style.setProperty('--ring-width', props.ringWidth * props.scale + 'px');
        loadingSpinner.value.style.setProperty('--width', props.width * props.scale + 'px');
        loadingSpinner.value.style.setProperty('--height', props.height * props.scale + 'px');
      }
    });
    return {
      loadingSpinner: loadingSpinner
    };
  }
});var _withId$9 = /*#__PURE__*/vue.withScopeId("data-v-04827584");

vue.pushScopeId("data-v-04827584");

var _hoisted_1$6 = {
  class: "loading-spinner-container"
};

var _hoisted_2$3 = /*#__PURE__*/vue.createVNode("span", {
  class: "hidden-dash"
}, "-", -1);

var _hoisted_3$1 = {
  class: "loading-spinner",
  ref: "loadingSpinner"
};

var _hoisted_4$1 = /*#__PURE__*/vue.createVNode("div", {
  class: "lds-dual-ring"
}, null, -1);

var _hoisted_5$1 = /*#__PURE__*/vue.createVNode("span", {
  class: "hidden-dash"
}, "-", -1);

vue.popScopeId();

var render$9 = /*#__PURE__*/_withId$9(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", _hoisted_1$6, [_hoisted_2$3, vue.createVNode("div", _hoisted_3$1, [_hoisted_4$1], 512), _hoisted_5$1]);
});var css_248z$b = "\n.loading-spinner-container[data-v-04827584] {\n    display: flex;\n    align-items: center;\n}\n.loading-spinner[data-v-04827584] {\n    display: inline-flex;\n    justify-content: center;\n    align-items: center;\n    position: relative;\n    width: var(--width);\n    height: var(--height);\n    min-width: var(--width);\n    min-height: var(--height);\n    overflow: hidden;\n}\n.lds-dual-ring[data-v-04827584] {\n    display: inline-flex;\n    justify-content: center;\n    align-items: center;\n    position: relative;\n    width: var(--width);\n    height: var(--height);\n    min-width: var(--width);\n    min-height: var(--height);\n}\n.lds-dual-ring[data-v-04827584]:before {\n    content: '';\n    position: absolute;\n    top: 0;\n    left: 0;\n    right: 0;\n    bottom: 0;\n    width: 100%;\n    height: 100%;\n    border-radius: 50%;\n    border: var(--ring-width) solid rgba(255, 255, 255, 0.05);\n    box-sizing: border-box;\n}\n.lds-dual-ring[data-v-04827584]:after {\n    content: '';\n    position: absolute;\n    top: 0;\n    left: 0;\n    right: 0;\n    bottom: 0;\n    width: 100%;\n    height: 100%;\n    border-radius: 50%;\n    border: var(--ring-width) solid var(--spinner-color);\n    border-color: var(--spinner-color) transparent transparent transparent;\n    animation: lds-dual-ring-04827584 1.2s linear infinite;\n    box-sizing: border-box;\n}\n@keyframes lds-dual-ring-04827584 {\n0% {\n        transform: rotate(0deg);\n}\n100% {\n        transform: rotate(360deg);\n}\n}\n.hidden-dash[data-v-04827584] {\n    user-select: none;\n    color: rgba(0, 0, 0, 0);\n}\n";
styleInject(css_248z$b);script$9.render = render$9;
script$9.__scopeId = "data-v-04827584";var script$a = vue.defineComponent({
  components: {
    VmLoadingSpinner: script$9
  },
  props: {
    config: {
      type: Object
    },
    rowData: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    editCallback: {
      type: Function
    },
    deleteCallback: {
      type: Function
    },
    isSelectable: {
      type: Boolean,
      default: false
    },
    isGridBorder: {
      type: Boolean,
      default: true
    },
    isLoading: {
      type: Boolean,
      default: false
    },
    maxRows: {
      type: Number,
      default: 10
    }
  },
  // eslint-disable-next-line
  setup: function setup(props) {
    // properties
    var selectedRowIds = vue.ref([]);
    var currentPage = vue.ref(1);
    var rowsPerPage = vue.computed(function () {
      return props.maxRows ? props.maxRows : 10;
    }); // computed

    var totalPages = vue.computed(function () {
      return Math.ceil(props.rowData.length / rowsPerPage.value);
    });
    var startRowIndex = vue.computed(function () {
      return Math.ceil((currentPage.value - 1) * rowsPerPage.value);
    });
    var endRowIndex = vue.computed(function () {
      var endRow = Math.ceil(currentPage.value * rowsPerPage.value - 1);

      if (endRow > props.rowData.length) {
        endRow = props.rowData.length - 1;
      }

      return endRow;
    });
    var emptyRows = vue.computed(function () {
      if (props.isLoading) {
        return 0;
      }

      var assign = 0;

      if (currentPage.value === totalPages.value && props.rowData.length % rowsPerPage.value !== 0) {
        assign = rowsPerPage.value - Math.floor(props.rowData.length % rowsPerPage.value);
      }

      return assign;
    }); // methods

    var selectRowById = function selectRowById(id) {
      if (props.isSelectable) {
        var index = selectedRowIds.value.indexOf(id);

        if (index !== -1) {
          selectedRowIds.value.splice(index, 1);
        } else {
          selectedRowIds.value.push(id);
        }
      }
    };

    var getSelectedIds = function getSelectedIds() {
      return selectedRowIds.value;
    }; // eslint-disable-next-line


    var getFilteredRows = function getFilteredRows() {
      if (props.isLoading) {
        return [];
      }

      var tmpRows = _toConsumableArray(props.rowData);

      return tmpRows.splice(startRowIndex.value, rowsPerPage.value);
    };

    var setCurrentPage = function setCurrentPage(pageIndex) {
      if (pageIndex > 0 && pageIndex <= totalPages.value) {
        currentPage.value = pageIndex;
      }
    };

    var incrementCurrentPage = function incrementCurrentPage() {
      setCurrentPage(currentPage.value + 1);
    };

    var decrementCurrentPage = function decrementCurrentPage() {
      setCurrentPage(currentPage.value - 1);
    };

    var goToLastPage = function goToLastPage() {
      setCurrentPage(totalPages.value);
    };

    var goToFirstPage = function goToFirstPage() {
      setCurrentPage(1);
    };

    var deselectAll = function deselectAll() {
      selectedRowIds.value = [];
    };

    return {
      props: props,
      selectRowById: selectRowById,
      selectedRowIds: selectedRowIds,
      getSelectedIds: getSelectedIds,
      rowsPerPage: rowsPerPage,
      currentPage: currentPage,
      totalPages: totalPages,
      startRowIndex: startRowIndex,
      endRowIndex: endRowIndex,
      emptyRows: emptyRows,
      getFilteredRows: getFilteredRows,
      incrementCurrentPage: incrementCurrentPage,
      decrementCurrentPage: decrementCurrentPage,
      goToLastPage: goToLastPage,
      goToFirstPage: goToFirstPage,
      deselectAll: deselectAll
    };
  }
});var _withId$a = /*#__PURE__*/vue.withScopeId("data-v-327da37c");

vue.pushScopeId("data-v-327da37c");

var _hoisted_1$7 = {
  class: "grid-wrapper"
};
var _hoisted_2$4 = {
  key: 0,
  style: {
    "width": "1px"
  }
};
var _hoisted_3$2 = {
  key: 0,
  class: "loading-row"
};
var _hoisted_4$2 = {
  colspan: "10000"
};
var _hoisted_5$2 = {
  class: "row jc-center"
};
var _hoisted_6$1 = {
  key: 1
};

var _hoisted_7 = /*#__PURE__*/vue.createVNode("td", {
  class: "center-text",
  colspan: "10000"
}, " No data available ", -1);

var _hoisted_8 = {
  key: 0
};
var _hoisted_9 = {
  key: 0
};

var _hoisted_10 = /*#__PURE__*/vue.createVNode("td", null, ".", -1);

var _hoisted_11 = {
  class: "pagination"
};

var _hoisted_12 = /*#__PURE__*/vue.createVNode("div", {
  class: "row-counter"
}, null, -1);

var _hoisted_13 = {
  class: "page-controls"
};
var _hoisted_14 = {
  class: "current-page"
};

var _hoisted_15 = /*#__PURE__*/vue.createVNode("i", {
  class: "far fa-chevron-double-left"
}, null, -1);

var _hoisted_16 = /*#__PURE__*/vue.createVNode("i", {
  class: "far fa-chevron-left"
}, null, -1);

var _hoisted_17 = /*#__PURE__*/vue.createVNode("i", {
  class: "far fa-chevron-right"
}, null, -1);

var _hoisted_18 = /*#__PURE__*/vue.createVNode("i", {
  class: "far fa-chevron-double-right"
}, null, -1);

vue.popScopeId();

var render$a = /*#__PURE__*/_withId$a(function (_ctx, _cache, $props, $setup, $data, $options) {
  var _component_vm_loading_spinner = vue.resolveComponent("vm-loading-spinner");

  return vue.openBlock(), vue.createBlock("div", {
    class: ["grid-container", {
      'grid-border': _ctx.isGridBorder
    }]
  }, [vue.createVNode("div", _hoisted_1$7, [vue.createVNode("table", {
    class: ["grid", {
      'is-selectable': _ctx.props.isSelectable
    }]
  }, [vue.createVNode("thead", null, [vue.createVNode("tr", null, [_ctx.isSelectable ? (vue.openBlock(), vue.createBlock("th", _hoisted_2$4)) : vue.createCommentVNode("", true), (vue.openBlock(true), vue.createBlock(vue.Fragment, null, vue.renderList(_ctx.config, function (headerConfig, i) {
    return vue.openBlock(), vue.createBlock("th", {
      key: i
    }, vue.toDisplayString(headerConfig.title), 1);
  }), 128))])]), vue.createVNode("tbody", null, [_ctx.isLoading ? (vue.openBlock(), vue.createBlock("tr", _hoisted_3$2, [vue.createVNode("td", _hoisted_4$2, [vue.createVNode("div", _hoisted_5$2, [vue.createVNode(_component_vm_loading_spinner, {
    scale: 0.8
  }, null, 8, ["scale"])])])])) : vue.createCommentVNode("", true), _ctx.rowData.length === 0 && !_ctx.isLoading ? (vue.openBlock(), vue.createBlock("tr", _hoisted_6$1, [_hoisted_7])) : vue.createCommentVNode("", true), (vue.openBlock(true), vue.createBlock(vue.Fragment, null, vue.renderList(_ctx.getFilteredRows(), function (row, i) {
    return vue.openBlock(), vue.createBlock("tr", {
      class: {
        selected: _ctx.selectedRowIds.indexOf(row.id) !== -1
      },
      key: i
    }, [_ctx.isSelectable ? (vue.openBlock(), vue.createBlock("td", _hoisted_8, [vue.createVNode("input", {
      type: "checkbox",
      checked: _ctx.selectedRowIds.find(function (x) {
        return x === row.id;
      }),
      onChange: function onChange($event) {
        return _ctx.selectRowById(row.id);
      }
    }, null, 40, ["checked", "onChange"])])) : vue.createCommentVNode("", true), (vue.openBlock(true), vue.createBlock(vue.Fragment, null, vue.renderList(_ctx.config, function (headerConfig, i) {
      return vue.openBlock(), vue.createBlock("td", {
        class: {
          'fit-width': headerConfig.fitWidth
        },
        key: i
      }, [!headerConfig.cellRenderer ? (vue.openBlock(), vue.createBlock("div", _hoisted_9, vue.toDisplayString(row[headerConfig.field]), 1)) : vue.createCommentVNode("", true), headerConfig.cellRenderer ? (vue.openBlock(), vue.createBlock(vue.resolveDynamicComponent(headerConfig.cellRenderer.component), {
        key: 1,
        row: row,
        field: headerConfig.field,
        exported: headerConfig.cellRenderer.exported
      }, null, 8, ["row", "field", "exported"])) : vue.createCommentVNode("", true)], 2);
    }), 128))], 2);
  }), 128)), (vue.openBlock(true), vue.createBlock(vue.Fragment, null, vue.renderList(_ctx.emptyRows, function (i) {
    return vue.openBlock(), vue.createBlock("tr", {
      class: "empty-row",
      key: i
    }, [_hoisted_10]);
  }), 128))])], 2)]), vue.createVNode("div", _hoisted_11, [vue.createVNode("div", {
    class: ["pagination-wrapper", {
      'pagination-hidden': _ctx.isLoading
    }]
  }, [_hoisted_12, vue.createVNode("div", _hoisted_13, [vue.createVNode("div", _hoisted_14, vue.toDisplayString(_ctx.rowData.length ? _ctx.startRowIndex + 1 : 0) + "-" + vue.toDisplayString(_ctx.endRowIndex + 1) + " of " + vue.toDisplayString(_ctx.rowData.length), 1), vue.createVNode("div", {
    class: "page-control-button",
    onClick: _cache[1] || (_cache[1] = function () {
      return _ctx.goToFirstPage && _ctx.goToFirstPage.apply(_ctx, arguments);
    })
  }, [_hoisted_15]), vue.createVNode("div", {
    class: "page-control-button",
    onClick: _cache[2] || (_cache[2] = function () {
      return _ctx.decrementCurrentPage && _ctx.decrementCurrentPage.apply(_ctx, arguments);
    })
  }, [_hoisted_16]), vue.createVNode("div", {
    class: "page-control-button",
    onClick: _cache[3] || (_cache[3] = function () {
      return _ctx.incrementCurrentPage && _ctx.incrementCurrentPage.apply(_ctx, arguments);
    })
  }, [_hoisted_17]), vue.createVNode("div", {
    class: "page-control-button",
    onClick: _cache[4] || (_cache[4] = function () {
      return _ctx.goToLastPage && _ctx.goToLastPage.apply(_ctx, arguments);
    })
  }, [_hoisted_18])])], 2)])], 2);
});var css_248z$c = "\n.grid-container[data-v-327da37c] {\n    --grid-border-color: rgba(var(--default-color), 0.3);\n    overflow: hidden;\n    font-size: 0.9rem;\n    flex: 1;\n}\n.grid-wrapper[data-v-327da37c] {\n    overflow-x: auto;\n}\n.grid[data-v-327da37c] {\n    border-collapse: collapse;\n    margin: 0 auto;\n    overflow: hidden;\n    overflow-x: scroll;\n    width: 100%;\n}\n.grid thead tr[data-v-327da37c] {\n    color: rgb(var(--default-color));\n    text-align: left;\n    border-bottom: 1px solid var(--grid-border-color);\n}\n.grid thead tr th[data-v-327da37c] {\n    font-weight: 600;\n}\n.grid th[data-v-327da37c] {\n    white-space: nowrap;\n    padding: 0 16px;\n    height: 50px;\n}\n.grid td[data-v-327da37c] {\n    height: 46px;\n    white-space: nowrap;\n}\n.grid td[data-v-327da37c] {\n    padding: 0 16px;\n}\n.grid tbody tr[data-v-327da37c] {\n    border-bottom: 1px solid rgba(var(--default-color), 0.1);\n    transition-duration: 0.1s;\n}\n.grid tbody tr[data-v-327da37c]:last-child {\n    border-bottom: none;\n}\n.grid tbody tr[data-v-327da37c]:hover {\n    background: rgba(255, 255, 255, 0.02);\n    transition-duration: 0.1s;\n}\n.grid.is-selectable tbody tr.selected[data-v-327da37c] {\n    background: #d5eaf8;\n}\n.grid tbody tr.empty-row[data-v-327da37c] {\n    visibility: hidden;\n    border-bottom: 1px solid rgba(0, 0, 0, 0);\n}\n.grid tbody tr.empty-row[data-v-327da37c]:last-child {\n    border-bottom: none;\n}\n.grid tbody tr td.fit-width[data-v-327da37c] {\n    width: 1px;\n    white-space: nowrap;\n}\n.pagination-hidden[data-v-327da37c] {\n    visibility: hidden;\n    pointer-events: none;\n    user-select: none;\n}\n.pagination[data-v-327da37c] {\n    padding: 10px 10px;\n    color: rgb(var(--default-color));\n    border-top: 1px solid var(--grid-border-color);\n}\n.pagination-wrapper[data-v-327da37c] {\n    display: flex;\n    justify-content: space-between;\n}\n.pagination > .pagination-wrapper > .page-controls[data-v-327da37c] {\n    display: flex;\n}\n.pagination > .pagination-wrapper > .page-controls > .current-page[data-v-327da37c] {\n    padding: 0 10px;\n}\n.pagination > .pagination-wrapper > .page-controls > .page-control-button[data-v-327da37c] {\n    color: rgb(var(--default-color), 0.4);\n    padding: 0 3px;\n    margin: 0 10px;\n    cursor: pointer;\n    transition-duration: 0.1s;\n}\n.pagination\n    > .pagination-wrapper\n    > .page-controls\n    > .page-control-button[data-v-327da37c]:hover {\n    color: rgb(var(--default-color), 0.2);\n    transition-duration: 0.1s;\n}\n.grid tbody tr.loading-row[data-v-327da37c] {\n    border-bottom: none;\n}\n.grid tbody tr.loading-row td[data-v-327da37c] {\n    padding: 10px 0;\n}\n";
styleInject(css_248z$c);script$a.render = render$a;
script$a.__scopeId = "data-v-327da37c";// vue
var script$b = vue.defineComponent({
  components: {
    VmActionButton: script,
    VmCard: script$3
  },
  props: {
    title: {
      type: String,
      default: ''
    },
    isPreventClose: {
      type: Boolean,
      default: false
    }
  },
  // eslint-disable-next-line
  setup: function setup(props) {
    // properties
    var isVisible = vue.ref(false); // methods

    var show = function show() {
      isVisible.value = true;
    };

    var hide = function hide() {
      if (!props.isPreventClose) {
        isVisible.value = false;
      }
    };

    return {
      props: props,
      isVisible: isVisible,
      show: show,
      hide: hide
    };
  }
});var _withId$b = /*#__PURE__*/vue.withScopeId("data-v-b946818e");

vue.pushScopeId("data-v-b946818e");

var _hoisted_1$8 = {
  class: "modal"
};
var _hoisted_2$5 = {
  class: "modal-content"
};

vue.popScopeId();

var render$b = /*#__PURE__*/_withId$b(function (_ctx, _cache, $props, $setup, $data, $options) {
  var _component_vm_card = vue.resolveComponent("vm-card");

  return vue.openBlock(), vue.createBlock(vue.Teleport, {
    to: "#app-theme"
  }, [vue.createVNode("div", _hoisted_1$8, [vue.createVNode(_component_vm_card, {
    class: ["modal-pane", {
      'modal-visible': _ctx.isVisible
    }],
    elevation: 8
  }, {
    default: _withId$b(function () {
      return [vue.createVNode("div", _hoisted_2$5, [vue.renderSlot(_ctx.$slots, "default")])];
    }),
    _: 3
  }, 8, ["class"]), vue.createVNode("div", {
    class: ["modal-background", {
      'modal-background-visible': _ctx.isVisible
    }],
    onClick: _cache[1] || (_cache[1] = function ($event) {
      return _ctx.hide();
    })
  }, null, 2)])]);
});var css_248z$d = "\n.modal[data-v-b946818e] {\n    position: fixed;\n    top: 0;\n    bottom: 0;\n    left: 0;\n    right: 0;\n    display: flex;\n    justify-content: center;\n    align-items: center;\n    z-index: 999;\n    pointer-events: none;\n}\n.modal-content[data-v-b946818e] {\n    height: 100%;\n    overflow-y: auto;\n}\n.modal-pane[data-v-b946818e] {\n    position: relative;\n    z-index: 1000;\n    overflow: hidden;\n    border-radius: 3px;\n    width: 600px;\n    max-width: 600px;\n    margin: 30px;\n    display: flex;\n    flex-direction: column;\n    max-height: 90%;\n    opacity: 0;\n    pointer-events: none;\n    z-index: 2;\n    transform: scale(0.8);\n    transition: transform 0.2s, opacity 0.2s;\n}\n.modal-background[data-v-b946818e] {\n    opacity: 0;\n    pointer-events: none;\n    position: fixed;\n    background: rgba(0, 0, 0, 0.5);\n    top: 0;\n    bottom: 0;\n    left: 0;\n    right: 0;\n    z-index: 1;\n    transition: opacity 0.2s;\n}\n.modal-visible[data-v-b946818e] {\n    transform: scale(1);\n}\n.modal-background-visible[data-v-b946818e],\n.modal-visible[data-v-b946818e] {\n    opacity: 1;\n    pointer-events: all;\n    transition: transform 0.2s, opacity 0.2s;\n}\n";
styleInject(css_248z$d);script$b.render = render$b;
script$b.__scopeId = "data-v-b946818e";// vue
var script$c = vue.defineComponent({
  props: {
    elevation: {
      type: Number,
      default: 2
    }
  },
  setup: function setup(props) {
    return {
      props: props
    };
  }
});var _withId$c = /*#__PURE__*/vue.withScopeId("data-v-0359d7af");

var render$c = /*#__PURE__*/_withId$c(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", {
    class: ["navigation-drawer", ["elevation-".concat(_ctx.props.elevation), "elevation-background-".concat(_ctx.props.elevation)]]
  }, [vue.renderSlot(_ctx.$slots, "default")], 2);
});var css_248z$e = "\n.navigation-drawer[data-v-0359d7af] {\n    position: sticky;\n    top: 4rem;\n    height: calc(100vh - 4rem);\n    padding: 20px 32px;\n    overflow-y: auto;\n}\n";
styleInject(css_248z$e);script$c.render = render$c;
script$c.__scopeId = "data-v-0359d7af";// vue
var script$d = vue.defineComponent({
  props: {
    modelValue: {
      type: Number,
      default: 0,
      required: true
    },
    label: {
      type: String,
      default: 'Default'
    }
  },
  // eslint-disable-next-line
  setup: function setup(props, context) {
    // refs
    var textField = vue.ref(null); // computed

    var isInputFilled = vue.computed(function () {
      return (props.modelValue + '').length > 0;
    }); // methods

    var updateValue = function updateValue(e) {
      var val = e.target.value;

      try {
        parseInt(val);
        context.emit('update:modelValue', val);
      } catch (e) {
        window.console.log(e);
      }
    };

    return {
      props: props,
      isInputFilled: isInputFilled,
      textField: textField,
      updateValue: updateValue
    };
  }
});var _withId$d = /*#__PURE__*/vue.withScopeId("data-v-38e24d5a");

vue.pushScopeId("data-v-38e24d5a");

var _hoisted_1$9 = {
  class: "text-field"
};

var _hoisted_2$6 = /*#__PURE__*/vue.createVNode("div", {
  class: "text-field-left"
}, null, -1);

var _hoisted_3$3 = /*#__PURE__*/vue.createVNode("div", {
  class: "text-field-right"
}, null, -1);

vue.popScopeId();

var render$d = /*#__PURE__*/_withId$d(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", _hoisted_1$9, [vue.createVNode("input", {
    class: "text-field-input",
    type: "text",
    ref: "textField",
    value: _ctx.modelValue,
    onInput: _cache[1] || (_cache[1] = function () {
      return _ctx.updateValue && _ctx.updateValue.apply(_ctx, arguments);
    })
  }, null, 40, ["value"]), _hoisted_2$6, vue.createVNode("div", {
    class: ["text-field-notch", {
      'text-field-notch-filled': _ctx.isInputFilled
    }]
  }, [vue.createVNode("div", {
    class: ["text-field-label", {
      'text-field-label-filled': _ctx.isInputFilled
    }]
  }, vue.toDisplayString(_ctx.props.label), 3)], 2), _hoisted_3$3]);
});var css_248z$f = "\n.text-field[data-v-38e24d5a] {\n    --font-size: 1.1rem;\n    --label-padding: 6px;\n    --horizontal-padding: 16px;\n    --vertical-padding: 16px;\n    --border: 1px solid rgba(255, 255, 255, 0.3);\n    --height: calc(var(--font-size) + (var(--vertical-padding) * 2) + 2px);\n    display: flex;\n    align-items: center;\n    width: 100%;\n    position: relative;\n}\n.text-field-left[data-v-38e24d5a],\n.text-field-notch[data-v-38e24d5a],\n.text-field-right[data-v-38e24d5a] {\n    min-height: var(--height);\n    max-height: var(--height);\n    padding: var(--vertical-padding) 0;\n}\n.text-field-left[data-v-38e24d5a] {\n    padding-left: calc(var(--horizontal-padding) - var(--label-padding));\n    border: var(--border);\n    border-right: none;\n    border-top-left-radius: 4px;\n    border-bottom-left-radius: 4px;\n}\n.text-field-notch[data-v-38e24d5a] {\n    display: flex;\n    align-items: center;\n    padding: var(--vertical-padding) 0;\n    border-top: var(--border);\n    border-bottom: var(--border);\n}\n.text-field-right[data-v-38e24d5a] {\n    width: 100%;\n    border: var(--border);\n    border-left: none;\n    border-top-right-radius: 4px;\n    border-bottom-right-radius: 4px;\n}\n.text-field input[data-v-38e24d5a] {\n    flex: 1;\n    position: absolute;\n    padding: var(--vertical-padding) var(--horizontal-padding);\n    outline: none;\n    border: none;\n    border-radius: 5px;\n    background: rgba(0, 0, 0, 0);\n    color: #fff;\n    font-size: var(--font-size);\n    width: 100%;\n}\n.text-field-label[data-v-38e24d5a] {\n    flex: 1;\n    padding: 0 6px;\n    color: #aaa;\n    transition-duration: 0.2s;\n    white-space: nowrap;\n    font-size: var(--font-size);\n}\ninput:focus ~ .text-field-notch[data-v-38e24d5a],\n.text-field-notch-filled[data-v-38e24d5a] {\n    border-top: none;\n}\ninput:focus ~ .text-field-notch .text-field-label[data-v-38e24d5a],\n.text-field-label-filled[data-v-38e24d5a] {\n    transform: translateY(calc((-1 / 2) * var(--height)));\n    font-size: 0.8rem;\n}\ninput:focus ~ .text-field-left[data-v-38e24d5a],\ninput:focus ~ .text-field-notch[data-v-38e24d5a],\ninput:focus ~ .text-field-right[data-v-38e24d5a] {\n    border-color: #2c60d1;\n}\ninput:focus ~ .text-field-notch .text-field-label[data-v-38e24d5a] {\n    color: #82a4ed;\n}\n";
styleInject(css_248z$f);script$d.render = render$d;
script$d.__scopeId = "data-v-38e24d5a";// vue
var script$e = vue.defineComponent({
  setup: function setup() {
    return {};
  }
});var _withId$e = /*#__PURE__*/vue.withScopeId("data-v-7f859afc");

vue.pushScopeId("data-v-7f859afc");

var _hoisted_1$a = {
  class: "page-header"
};

vue.popScopeId();

var render$e = /*#__PURE__*/_withId$e(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", _hoisted_1$a, [vue.renderSlot(_ctx.$slots, "default")]);
});var css_248z$g = "\n.page-header[data-v-7f859afc] {\n    font-weight: 500;\n    font-size: 1.8rem;\n    margin-top: calc(var(--default-margin) * 4);\n    margin-bottom: calc(var(--default-margin) * 6);\n}\n";
styleInject(css_248z$g);script$e.render = render$e;
script$e.__scopeId = "data-v-7f859afc";var script$f = {
  props: {
    id: {
      type: Number
    },
    modelValue: {
      type: Boolean,
      default: false
    },
    label: {
      type: String
    }
  },
  // eslint-disable-next-line
  setup: function setup(props, context) {
    // methods
    // eslint-disable-next-line
    var updateValue = function updateValue(e) {
      context.emit('update:modelValue', e.target.checked);
    };

    return {
      props: props,
      updateValue: updateValue
    };
  }
};var _withId$f = /*#__PURE__*/vue.withScopeId("data-v-65487313");

vue.pushScopeId("data-v-65487313");

var _hoisted_1$b = {
  class: "switch-container"
};
var _hoisted_2$7 = {
  class: "switch"
};

var _hoisted_3$4 = /*#__PURE__*/vue.createVNode("span", {
  class: "slider"
}, null, -1);

var _hoisted_4$3 = {
  key: 0,
  class: "label"
};

vue.popScopeId();

var render$f = /*#__PURE__*/_withId$f(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", _hoisted_1$b, [vue.createVNode("label", _hoisted_2$7, [vue.createVNode("input", {
    id: $props.id,
    type: "checkbox",
    checked: $props.modelValue,
    onInput: _cache[1] || (_cache[1] = function () {
      return $setup.updateValue && $setup.updateValue.apply($setup, arguments);
    })
  }, null, 40, ["id", "checked"]), _hoisted_3$4]), $setup.props.label ? (vue.openBlock(), vue.createBlock("label", _hoisted_4$3, vue.toDisplayString($setup.props.label), 1)) : vue.createCommentVNode("", true)]);
});var css_248z$h = "\n.switch-container[data-v-65487313] {\n    display: flex;\n    align-items: center;\n}\n.switch[data-v-65487313] {\n    --width: 34px;\n    --switch-size: 20px;\n    --height: 14px;\n    --border-radius: 100px;\n    --switch-border-radius: 50%;\n    --inner-padding: -1px;\n\n    width: var(--width);\n    min-width: var(--width);\n    height: var(--height);\n    cursor: pointer;\n    background-color: #adadad;\n    border-radius: var(--border-radius);\n    position: relative;\n    display: inline-block;\n}\n.switch input[data-v-65487313] {\n    display: none;\n}\n.slider[data-v-65487313] {\n    position: absolute;\n    border-radius: inherit;\n    top: 0;\n    left: 0;\n    right: 0;\n    bottom: 0;\n    transition: 0.4s;\n    z-index: 1;\n}\n.slider[data-v-65487313]:before {\n    display: flex;\n    justify-content: center;\n    position: absolute;\n    border-radius: var(--switch-border-radius);\n    content: '';\n    width: var(--switch-size);\n    height: var(--switch-size);\n    left: var(--inner-padding);\n    top: calc((var(--height) / 2) - (var(--switch-size) / 2));\n    background-color: #fff;\n    transition: 0.3s;\n    box-shadow: var(--elevation-2);\n}\ninput:checked + .slider[data-v-65487313] {\n    background: rgb(177, 127, 247);\n    border-radius: inherit;\n}\n.slider[data-v-65487313]:after {\n    --padding: 12px;\n    content: '';\n    background: rgb(177, 127, 247, 0);\n    border-radius: 200px;\n    display: flex;\n    width: var(--switch-size);\n    height: var(--switch-size);\n    left: calc(var(--inner-padding) - var(--padding));\n    top: calc((var(--height) / 2) - (var(--switch-size) / 2) - var(--padding));\n    position: absolute;\n    padding: var(--padding);\n    pointer-events: none;\n    z-index: -1;\n    transition: background 0.2s, transform 0.3s;\n}\ninput:hover + .slider[data-v-65487313]:after {\n    background: rgb(173, 173, 173, 0.2);\n}\ninput:checked + .slider[data-v-65487313]:after {\n    transform: translateX(\n        calc(var(--width) - var(--switch-size) - (2 * var(--inner-padding)))\n    );\n}\ninput:checked:hover + .slider[data-v-65487313]:after {\n    background: rgb(177, 127, 247, 0.2);\n}\ninput:checked ~ .slider[data-v-65487313]:before {\n    background: rgb(98, 0, 238);\n}\ninput:focus + .slider[data-v-65487313] {\n}\ninput:checked + .slider[data-v-65487313]:before {\n    transform: translateX(\n        calc(var(--width) - var(--switch-size) - (2 * var(--inner-padding)))\n    );\n}\n.label[data-v-65487313] {\n    margin-left: calc(var(--default-margin) * 3);\n}\n";
styleInject(css_248z$h);script$f.render = render$f;
script$f.__scopeId = "data-v-65487313";// vue
var script$g = vue.defineComponent({
  props: {
    modelValue: {
      type: String
    },
    variant: {
      type: String,
      default: 'filled'
    },
    label: {
      type: String,
      default: 'Default'
    },
    isPassword: {
      type: Boolean,
      default: false
    }
  },
  // eslint-disable-next-line
  setup: function setup(props, context) {
    // refs
    var textField = vue.ref(null); // computed

    var isInputFilled = vue.ref(false); // methods

    var updateValue = function updateValue(e) {
      if (textField.value) {
        isInputFilled.value = textField.value.value.length > 0;
      }

      if (props.modelValue !== undefined) {
        var val = e.target.value;
        context.emit('update:modelValue', val);
      }
    };

    return {
      props: props,
      isInputFilled: isInputFilled,
      textField: textField,
      updateValue: updateValue
    };
  }
});var _withId$g = /*#__PURE__*/vue.withScopeId("data-v-3297fe0e");

vue.pushScopeId("data-v-3297fe0e");

var _hoisted_1$c = /*#__PURE__*/vue.createVNode("div", {
  class: "text-field-left"
}, null, -1);

var _hoisted_2$8 = /*#__PURE__*/vue.createVNode("div", {
  class: "text-field-right"
}, null, -1);

vue.popScopeId();

var render$g = /*#__PURE__*/_withId$g(function (_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", {
    class: ["text-field", 'text-field-' + _ctx.props.variant]
  }, [_ctx.modelValue !== undefined ? (vue.openBlock(), vue.createBlock("input", {
    key: 0,
    class: "text-field-input",
    type: _ctx.isPassword ? 'password' : 'text',
    ref: "textField",
    value: _ctx.modelValue,
    onInput: _cache[1] || (_cache[1] = function () {
      return _ctx.updateValue && _ctx.updateValue.apply(_ctx, arguments);
    })
  }, null, 40, ["type", "value"])) : vue.createCommentVNode("", true), _ctx.modelValue === undefined ? (vue.openBlock(), vue.createBlock("input", {
    key: 1,
    class: "text-field-input",
    type: _ctx.isPassword ? 'password' : 'text2',
    ref: "textField",
    onInput: _cache[2] || (_cache[2] = function () {
      return _ctx.updateValue && _ctx.updateValue.apply(_ctx, arguments);
    })
  }, null, 40, ["type"])) : vue.createCommentVNode("", true), _hoisted_1$c, vue.createVNode("div", {
    class: ["text-field-notch", {
      'text-field-notch-filled': _ctx.isInputFilled
    }]
  }, [vue.createVNode("div", {
    class: ["text-field-label", {
      'text-field-label-filled': _ctx.isInputFilled
    }]
  }, vue.toDisplayString(_ctx.props.label), 3)], 2), _hoisted_2$8], 2);
});var css_248z$i = "\n.text-field[data-v-3297fe0e] {\n    --font-size: 1rem;\n    --label-padding: 6px;\n    --horizontal-padding: 16px;\n    --border: 1px solid rgba(var(--default-color), 0.42);\n    --height: calc(var(--font-size) + (var(--padding-top) + var(--padding-bottom)) + 2px);\n    display: flex;\n    align-items: center;\n    width: 100%;\n    position: relative;\n}\n.text-field-filled[data-v-3297fe0e] {\n    --padding-top: 24px;\n    --padding-bottom: calc(38px - var(--padding-top));\n    transition: background 0.2s;\n}\n.text-field-filled[data-v-3297fe0e]:hover {\n    --padding-bottom: calc(38px - var(--padding-top));\n}\n.text-field-outlined[data-v-3297fe0e] {\n    --padding-top: 18px;\n    --padding-bottom: 18px;\n}\n.text-field-outlined:hover .text-field-left[data-v-3297fe0e],\n.text-field-outlined:hover .text-field-notch[data-v-3297fe0e],\n.text-field-outlined:hover .text-field-right[data-v-3297fe0e] {\n    --border: 1px solid rgba(var(--default-color), 0.87);\n}\n.text-field-filled .text-field-left[data-v-3297fe0e],\n.text-field-filled .text-field-notch[data-v-3297fe0e],\n.text-field-filled .text-field-right[data-v-3297fe0e] {\n    background: rgba(var(--default-color), 0.03);\n}\n.text-field-filled:hover .text-field-left[data-v-3297fe0e],\n.text-field-filled:hover .text-field-notch[data-v-3297fe0e],\n.text-field-filled:hover .text-field-right[data-v-3297fe0e] {\n    background: rgba(var(--default-color), 0.06);\n    --border: 1px solid rgba(var(--default-color), 0.87);\n}\n.text-field-left[data-v-3297fe0e],\n.text-field-notch[data-v-3297fe0e],\n.text-field-right[data-v-3297fe0e] {\n    min-height: 100%;\n    max-height: 100%;\n    height: 100%;\n}\n.text-field-left[data-v-3297fe0e],\n.text-field-right[data-v-3297fe0e] {\n    min-height: var(--height);\n    max-height: var(--height);\n    padding-top: var(--padding-top);\n    padding-bottom: var(--padding-bottom);\n}\n.text-field-notch[data-v-3297fe0e] {\n    min-height: var(--height);\n    max-height: var(--height);\n    padding-top: 18px;\n    padding-bottom: 18px;\n}\n.text-field-left[data-v-3297fe0e] {\n    border-bottom: var(--border);\n    padding-left: calc(var(--horizontal-padding) - var(--label-padding));\n    border-top-left-radius: 4px;\n}\n.text-field-outlined .text-field-left[data-v-3297fe0e] {\n    border: var(--border);\n    border-right: none;\n    border-bottom-left-radius: 4px;\n}\n.text-field-notch[data-v-3297fe0e] {\n    display: flex;\n    align-items: center;\n    border-bottom: var(--border);\n}\n.text-field-outlined .text-field-notch[data-v-3297fe0e] {\n    border-top: var(--border);\n}\n.text-field-right[data-v-3297fe0e] {\n    border-bottom: var(--border);\n    width: 100%;\n    border-top-right-radius: 4px;\n}\n.text-field-outlined .text-field-right[data-v-3297fe0e] {\n    border: var(--border);\n    border-left: none;\n    border-bottom-right-radius: 4px;\n}\n.text-field input[data-v-3297fe0e] {\n    flex: 1;\n    position: absolute;\n    padding: 0 var(--horizontal-padding);\n    padding-top: var(--padding-top);\n    padding-bottom: var(--padding-bottom);\n    outline: none;\n    border: none;\n    border-radius: 5px;\n    background: rgba(0, 0, 0, 0);\n    color: var(--default-color);\n    font-size: var(--font-size);\n    width: 100%;\n}\n.text-field-label[data-v-3297fe0e] {\n    flex: 1;\n    padding: 0 6px;\n    color: rgba(var(--default-color), 0.6);\n    transition-duration: 0.2s;\n    white-space: nowrap;\n    font-size: var(--font-size);\n}\n.text-field-outlined input:focus ~ .text-field-notch[data-v-3297fe0e],\n.text-field-outlined .text-field-notch-filled[data-v-3297fe0e] {\n    border-top: none;\n}\n.text-field-filled input:focus ~ .text-field-notch .text-field-label[data-v-3297fe0e],\n.text-field-filled .text-field-label-filled[data-v-3297fe0e] {\n    transform: translateY(calc((-1 / 4) * var(--height)));\n    font-size: 0.8rem;\n}\n.text-field-outlined input:focus ~ .text-field-notch .text-field-label[data-v-3297fe0e],\n.text-field-outlined .text-field-label-filled[data-v-3297fe0e] {\n    transform: translateY(calc((-1 / 2) * var(--height)));\n    font-size: 0.8rem;\n}\n.text-field-filled input:focus ~ .text-field-left[data-v-3297fe0e],\n.text-field-filled input:focus ~ .text-field-notch[data-v-3297fe0e],\n.text-field-filled input:focus ~ .text-field-right[data-v-3297fe0e] {\n    background: rgba(var(--default-color), 0.1);\n}\ninput:focus ~ .text-field-left[data-v-3297fe0e],\ninput:focus ~ .text-field-notch[data-v-3297fe0e],\ninput:focus ~ .text-field-right[data-v-3297fe0e] {\n    border-color: #2c60d1;\n}\ninput:focus ~ .text-field-notch .text-field-label[data-v-3297fe0e] {\n    color: #2c60d1;\n}\ninput[data-v-3297fe0e]:-webkit-autofill,\ninput[data-v-3297fe0e]:-webkit-autofill:hover,\ninput[data-v-3297fe0e]:-webkit-autofill:focus,\ninput[data-v-3297fe0e]:-webkit-autofill:active {\n    -webkit-background-clip: text;\n    -webkit-text-fill-color: rgb(var(--default-color));\n}\n";
styleInject(css_248z$i);script$g.render = render$g;
script$g.__scopeId = "data-v-3297fe0e";var script$h = {
  props: {
    mode: {
      type: String,
      default: 'light'
    }
  },
  // eslint-disable-next-line
  setup: function setup(props) {
    return {
      props: props
    };
  }
};function render$h(_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createBlock("div", {
    id: "app-theme",
    class: ["theme", $setup.props.mode + '-theme'],
    ref: "theme"
  }, [vue.renderSlot(_ctx.$slots, "default")], 2);
}var css_248z$j = "\n@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700;900&display=swap');\n* {\n    box-sizing: border-box;\n}\nhtml {\n    display: flex;\n    width: 100%;\n    min-height: 100%;\n}\nbody {\n    display: flex;\n    width: 100%;\n    height: 100%;\n    flex: 1;\n    margin: 0;\n}\n#app {\n    display: flex;\n    width: 100%;\n    height: 100%;\n    flex: 1;\n}\n.theme {\n    --default-margin: 5px;\n    width: 100%;\n    height: 100%;\n    flex: 1;\n    font-family: 'Roboto', sans-serif;\n    background: rgb(var(--default-background));\n    color: rgb(var(--default-color));\n\n    --baseline-color: 0, 0, 0;\n    --umbra-opacity: 0.2;\n    --penumbra-opacity: 0.14;\n    --ambient-opacity: 0.12;\n\n    --umbra-map-0: 0px 0px 0px 0px;\n    --umbra-map-1: 0px 2px 1px -1px;\n    --umbra-map-2: 0px 3px 1px -2px;\n    --umbra-map-3: 0px 3px 3px -2px;\n    --umbra-map-4: 0px 2px 4px -1px;\n    --umbra-map-5: 0px 3px 5px -1px;\n    --umbra-map-6: 0px 3px 5px -1px;\n    --umbra-map-7: 0px 4px 5px -2px;\n    --umbra-map-8: 0px 5px 5px -3px;\n    --umbra-map-9: 0px 5px 6px -3px;\n    --umbra-map-10: 0px 6px 6px -3px;\n    --umbra-map-11: 0px 6px 7px -4px;\n    --umbra-map-12: 0px 7px 8px -4px;\n    --umbra-map-13: 0px 7px 8px -4px;\n    --umbra-map-14: 0px 7px 9px -4px;\n    --umbra-map-15: 0px 8px 9px -5px;\n    --umbra-map-16: 0px 8px 10px -5px;\n    --umbra-map-17: 0px 8px 11px -5px;\n    --umbra-map-18: 0px 9px 11px -5px;\n    --umbra-map-19: 0px 9px 12px -6px;\n    --umbra-map-20: 0px 10px 13px -6px;\n    --umbra-map-21: 0px 10px 13px -6px;\n    --umbra-map-22: 0px 10px 14px -6px;\n    --umbra-map-23: 0px 11px 14px -7px;\n    --umbra-map-24: 0px 11px 15px -7px;\n    --penumbra-map-0: 0px 0px 0px 0px;\n    --penumbra-map-1: 0px 1px 1px 0px;\n    --penumbra-map-2: 0px 2px 2px 0px;\n    --penumbra-map-3: 0px 3px 4px 0px;\n    --penumbra-map-4: 0px 4px 5px 0px;\n    --penumbra-map-5: 0px 5px 8px 0px;\n    --penumbra-map-6: 0px 6px 10px 0px;\n    --penumbra-map-7: 0px 7px 10px 1px;\n    --penumbra-map-8: 0px 8px 10px 1px;\n    --penumbra-map-9: 0px 9px 12px 1px;\n    --penumbra-map-10: 0px 10px 14px 1px;\n    --penumbra-map-11: 0px 11px 15px 1px;\n    --penumbra-map-12: 0px 12px 17px 2px;\n    --penumbra-map-13: 0px 13px 19px 2px;\n    --penumbra-map-14: 0px 14px 21px 2px;\n    --penumbra-map-15: 0px 15px 22px 2px;\n    --penumbra-map-16: 0px 16px 24px 2px;\n    --penumbra-map-17: 0px 17px 26px 2px;\n    --penumbra-map-18: 0px 18px 28px 2px;\n    --penumbra-map-19: 0px 19px 29px 2px;\n    --penumbra-map-20: 0px 20px 31px 3px;\n    --penumbra-map-21: 0px 21px 33px 3px;\n    --penumbra-map-22: 0px 22px 35px 3px;\n    --penumbra-map-23: 0px 23px 36px 3px;\n    --penumbra-map-24: 0px 24px 38px 3px;\n    --ambient-map-0: 0px 0px 0px 0px;\n    --ambient-map-1: 0px 1px 3px 0px;\n    --ambient-map-2: 0px 1px 5px 0px;\n    --ambient-map-3: 0px 1px 8px 0px;\n    --ambient-map-4: 0px 1px 10px 0px;\n    --ambient-map-5: 0px 1px 14px 0px;\n    --ambient-map-6: 0px 1px 18px 0px;\n    --ambient-map-7: 0px 2px 16px 1px;\n    --ambient-map-8: 0px 3px 14px 2px;\n    --ambient-map-9: 0px 3px 16px 2px;\n    --ambient-map-10: 0px 4px 18px 3px;\n    --ambient-map-11: 0px 4px 20px 3px;\n    --ambient-map-12: 0px 5px 22px 4px;\n    --ambient-map-13: 0px 5px 24px 4px;\n    --ambient-map-14: 0px 5px 26px 4px;\n    --ambient-map-15: 0px 6px 28px 5px;\n    --ambient-map-16: 0px 6px 30px 5px;\n    --ambient-map-17: 0px 6px 32px 5px;\n    --ambient-map-18: 0px 7px 34px 6px;\n    --ambient-map-19: 0px 7px 36px 6px;\n    --ambient-map-20: 0px 8px 38px 7px;\n    --ambient-map-21: 0px 8px 40px 7px;\n    --ambient-map-22: 0px 8px 42px 7px;\n    --ambient-map-23: 0px 9px 44px 8px;\n    --ambient-map-24: 0px 9px 46px 8px;\n    --elevation-0:\n        var(--umbra-map-0) rgba(var(--baseline-color) var(--umbra-opacity)),\n        var(--penumbra-map-0) rgba(var(--baseline-color) var(--penumbra-opacity)),\n        var(--ambient-map-0) rgba(var(--baseline-color) var(--ambient-opacity))\n    ;\n    --elevation-1:\n        var(--umbra-map-1) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-1) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-1) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-2:\n        var(--umbra-map-2) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-2) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-2) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-3:\n        var(--umbra-map-3) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-3) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-3) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-4:\n        var(--umbra-map-4) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-4) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-4) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-5:\n        var(--umbra-map-5) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-5) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-5) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-6:\n        var(--umbra-map-6) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-6) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-6) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-7:\n        var(--umbra-map-7) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-7) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-7) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-8:\n        var(--umbra-map-8) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-8) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-8) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-9:\n        var(--umbra-map-9) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-9) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-9) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-10:\n        var(--umbra-map-10) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-10) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-10) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-11:\n        var(--umbra-map-11) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-11) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-11) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-12:\n        var(--umbra-map-12) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-12) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-12) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-13:\n        var(--umbra-map-13) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-13) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-13) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-14:\n        var(--umbra-map-14) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-14) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-14) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-15:\n        var(--umbra-map-15) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-15) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-15) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-16:\n        var(--umbra-map-16) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-16) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-16) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-17:\n        var(--umbra-map-17) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-17) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-17) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-18:\n        var(--umbra-map-18) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-18) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-18) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-19:\n        var(--umbra-map-19) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-19) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-19) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-20:\n        var(--umbra-map-20) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-20) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-20) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-21:\n        var(--umbra-map-21) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-21) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-21) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-22:\n        var(--umbra-map-22) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-22) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-22) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-23:\n        var(--umbra-map-23) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-23) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-23) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n    --elevation-24:\n        var(--umbra-map-24) rgba(var(--baseline-color), var(--umbra-opacity)),\n        var(--penumbra-map-24) rgba(var(--baseline-color), var(--penumbra-opacity)),\n        var(--ambient-map-24) rgba(var(--baseline-color), var(--ambient-opacity))\n    ;\n}\n.light-theme {\n    --default-surface: 255, 255, 255;\n    --default-background: 250, 250, 250;\n    --default-color: 0, 0, 0;\n}\n.dark-theme {\n    --default-surface: 24, 24, 24;\n    --default-background: 24, 24, 24;\n    --default-color: 255, 255, 255;\n    --overlay-color: 255, 255, 255;\n    --elevation-opacity-0: 0;\n    --elevation-opacity-1: 0.05;\n    --elevation-opacity-2: 0.07;\n    --elevation-opacity-3: 0.08;\n    --elevation-opacity-4: 0.09;\n    --elevation-opacity-5: 0.10;\n    --elevation-opacity-6: 0.11;\n    --elevation-opacity-7: 0.115;\n    --elevation-opacity-8: 0.12;\n    --elevation-opacity-9: 0.125;\n    --elevation-opacity-10: 0.13;\n    --elevation-opacity-11: 0.135;\n    --elevation-opacity-12: 0.14;\n    --elevation-opacity-13: 0.1425;\n    --elevation-opacity-14: 0.145;\n    --elevation-opacity-15: 0.1475;\n    --elevation-opacity-16: 0.15;\n    --elevation-opacity-17: 0.15125;\n    --elevation-opacity-18: 0.1525;\n    --elevation-opacity-19: 0.15375;\n    --elevation-opacity-20: 0.155;\n    --elevation-opacity-21: 0.15675;\n    --elevation-opacity-22: 0.1575;\n    --elevation-opacity-23: 0.15875;\n    --elevation-opacity-24: 0.16;\n}\n.elevation-background-0,\n.elevation-background-1,\n.elevation-background-2,\n.elevation-background-3,\n.elevation-background-4,\n.elevation-background-5,\n.elevation-background-6,\n.elevation-background-7,\n.elevation-background-8,\n.elevation-background-9,\n.elevation-background-10,\n.elevation-background-11,\n.elevation-background-12,\n.elevation-background-13,\n.elevation-background-14,\n.elevation-background-15,\n.elevation-background-16,\n.elevation-background-17,\n.elevation-background-18,\n.elevation-background-19,\n.elevation-background-20,\n.elevation-background-21,\n.elevation-background-22,\n.elevation-background-23,\n.elevation-background-24 {\n    position: relative;\n    background: rgb(var(--default-surface));\n}\n.elevation-background-0::before,\n.elevation-background-1::before,\n.elevation-background-2::before,\n.elevation-background-3::before,\n.elevation-background-4::before,\n.elevation-background-5::before,\n.elevation-background-6::before,\n.elevation-background-7::before,\n.elevation-background-8::before,\n.elevation-background-9::before,\n.elevation-background-10::before,\n.elevation-background-11::before,\n.elevation-background-12::before,\n.elevation-background-13::before,\n.elevation-background-14::before,\n.elevation-background-15::before,\n.elevation-background-16::before,\n.elevation-background-17::before,\n.elevation-background-18::before,\n.elevation-background-19::before,\n.elevation-background-20::before,\n.elevation-background-21::before,\n.elevation-background-22::before,\n.elevation-background-23::before,\n.elevation-background-24::before {\n    content: '';\n    position: absolute;\n    top: 0;\n    left: 0;\n    right: 0;\n    bottom: 0;\n    pointer-events: none;\n}\n.elevation-background-0::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-0));\n}\n.elevation-background-1::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-1));\n}\n.elevation-background-2::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-2));\n}\n.elevation-background-3::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-3));\n}\n.elevation-background-4::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-4));\n}\n.elevation-background-5::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-5));\n}\n.elevation-background-6::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-6));\n}\n.elevation-background-7::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-7));\n}\n.elevation-background-8::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-8));\n}\n.elevation-background-9::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-9));\n}\n.elevation-background-10::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-10));\n}\n.elevation-background-11::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-11));\n}\n.elevation-background-12::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-12));\n}\n.elevation-background-13::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-13));\n}\n.elevation-background-14::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-14));\n}\n.elevation-background-15::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-15));\n}\n.elevation-background-16::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-16));\n}\n.elevation-background-17::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-17));\n}\n.elevation-background-18::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-18));\n}\n.elevation-background-19::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-19));\n}\n.elevation-background-20::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-20));\n}\n.elevation-background-21::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-21));\n}\n.elevation-background-22::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-22));\n}\n.elevation-background-23::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-23));\n}\n.elevation-background-24::before {\n    background: rgba(var(--overlay-color), var(--elevation-opacity-24));\n}\n.elevation-0 {\n    box-shadow: var(--elevation-0);\n}\n.elevation-1 {\n    box-shadow: var(--elevation-1);\n}\n.elevation-2 {\n    box-shadow: var(--elevation-2);\n}\n.elevation-3 {\n    box-shadow: var(--elevation-3);\n}\n.elevation-4 {\n    box-shadow: var(--elevation-4);\n}\n.elevation-5 {\n    box-shadow: var(--elevation-5);\n}\n.elevation-6 {\n    box-shadow: var(--elevation-6);\n}\n.elevation-7 {\n    box-shadow: var(--elevation-7);\n}\n.elevation-8 {\n    box-shadow: var(--elevation-8);\n}\n.elevation-9 {\n    box-shadow: var(--elevation-9);\n}\n.elevation-10 {\n    box-shadow: var(--elevation-10);\n}\n.elevation-11 {\n    box-shadow: var(--elevation-11);\n}\n.elevation-12 {\n    box-shadow: var(--elevation-12);\n}\n.elevation-13 {\n    box-shadow: var(--elevation-13);\n}\n.elevation-14 {\n    box-shadow: var(--elevation-14);\n}\n.elevation-15 {\n    box-shadow: var(--elevation-15);\n}\n.elevation-16 {\n    box-shadow: var(--elevation-16);\n}\n.elevation-17 {\n    box-shadow: var(--elevation-17);\n}\n.elevation-18 {\n    box-shadow: var(--elevation-18);\n}\n.elevation-19 {\n    box-shadow: var(--elevation-19);\n}\n.elevation-20 {\n    box-shadow: var(--elevation-20);\n}\n.elevation-21 {\n    box-shadow: var(--elevation-21);\n}\n.elevation-22 {\n    box-shadow: var(--elevation-22);\n}\n.elevation-23 {\n    box-shadow: var(--elevation-23);\n}\n.elevation-24 {\n    box-shadow: var(--elevation-24);\n}\n";
styleInject(css_248z$j);script$h.render = render$h;/* eslint-disable import/prefer-default-export */var components=/*#__PURE__*/Object.freeze({__proto__:null,VmActionButton: script,VmAppBar: script$1,VmButton: script$2,VmCard: script$3,VmCardContent: script$4,VmCardHeader: script$5,VmContainer: script$6,VmDivider: script$7,VmDropdown: script$8,VmGrid: script$a,VmLoadingSpinner: script$9,VmModal: script$b,VmNavigationDrawer: script$c,VmNumberField: script$d,VmPageHeader: script$e,VmSwitch: script$f,VmTextField: script$g,VmTheme: script$h});var install = function installVue3Material(app) {
  Object.entries(components).forEach(function (_ref) {
    var _ref2 = _slicedToArray(_ref, 2),
        componentName = _ref2[0],
        component = _ref2[1];

    app.component(componentName, component);
  });
}; // Create module definition for Vue.use()
var components$1=/*#__PURE__*/Object.freeze({__proto__:null,'default': install,VmActionButton: script,VmAppBar: script$1,VmButton: script$2,VmCard: script$3,VmCardContent: script$4,VmCardHeader: script$5,VmContainer: script$6,VmDivider: script$7,VmDropdown: script$8,VmGrid: script$a,VmLoadingSpinner: script$9,VmModal: script$b,VmNavigationDrawer: script$c,VmNumberField: script$d,VmPageHeader: script$e,VmSwitch: script$f,VmTextField: script$g,VmTheme: script$h});// only expose one global var, with component exports exposed as properties of
// that global var (eg. plugin.component)

Object.entries(components$1).forEach(function (_ref) {
  var _ref2 = _slicedToArray(_ref, 2),
      componentName = _ref2[0],
      component = _ref2[1];

  if (componentName !== 'default') {
    var key = componentName;
    var val = component;
    install[key] = val;
  }
});module.exports=install;
